akka-persistence/src/test/scala/akka/persistence/AtLeastOnceDeliveryCrashSpec.scala

```diff
@@ -72,7 +77,9 @@ class AtLeastOnceDeliveryCrashSpec
         system.actorOf(Props(new StoppingStrategySupervisor(testProbe.ref)), "supervisor")
       val superVisor = createCrashActorUnderSupervisor()
       superVisor ! CrashingActor.Message
-      testProbe.expectMsgType[CrashingActor.SendingMessage]
+      val msg = testProbe.expectMsgType[CrashingActor.SendingMessage]
+      // simulates confirmation from probe
+      superVisor ! CrashingActor.Confirmed(msg.deliveryId)
```

Comment:
A bit hard to follow test but a quick look at https://github.com/akka/akka/issues/16104 it covers that a message in flight (unconfirmed I guess) isn't replayed on recovery (or even stop). I think it _shouldn't_ confirm or that bugfix is no longer covered.

Reply:
I don't know if I follow your comment, but what I understand from this test is the following: 

1. we send two message: `CrashingActor.Message` and `CrashingActor.CrashMessage`
2. they won't crash the actor despite the name of the second message
3.`CrashingActor.Message` triggers a `deliver` call
4. actor is explicitly stopped
5. on recover, first message `CrashingActor.Message` is replayed and re-delivered because never confirmed
6. `CrashingActor.CrashMessage` is then replayed causing the actor to fail

I agree that my 'fix' is not helping here. As I understand it now, the test is bogus. 

Is my understand correct that what we should do instead is the following: 

1. send `CrashingActor.Message` and trigger a `deliver` (unconfirmed)
2. send `CrashingActor.CrashMessage` and crash the actor
3. restart the actor, replay `CrashingActor.Message` and confirm that the `deliver` is not retriggered

Is that how you understand it?

Reply:
Oh, now I see something else

The re-delivery should not happen during recovery, only after full recovery. Since we crash the actor during recovery, we should not have a redelivery. Now I see how the CrashMessage is trying to hold the re-delivery.